Build fails without deploy properties (base.deploy.dir) even for partial/module builds

### Summary
While setting up SW360 locally as a new contributor, I noticed that even partial or
module-level Maven builds fail unless deployment properties are provided.

### What I tried
I attempted a light / module build (e.g. building only `libraries`) without using the
deploy profile, expecting a minimal build to work for local verification.

### Actual behaviour
The build fails with the Maven Enforcer rule:


You must set at least the property 'base.deploy.dir'!

This happens even when building individual modules.

### Expected / Suggested improvement
It would help new contributors if:
- The requirement for `base.deploy.dir` (and related deploy properties) is clearly
  documented for *all* builds, including partial/module builds, **or**
- A minimal non-deploy build path is documented (if intended).

### Environment
- OS: Windows
- Java: 21 (Eclipse Adoptium)
- Maven: 3.9.x

Additional details:

Build attempted from command line (Git Bash / CMD).

Commands:
mvn -pl libraries -am package -DskipTests
mvn package -DskipTests -Dhelp-docs=false

Error:
"You must set at least the property 'base.deploy.dir'!"

Additional details:

Build attempted from command line (Git Bash / CMD).

Commands used:

1. mvn -pl libraries -am package -DskipTests
2. mvn package -DskipTests -Dhelp-docs=false

Resulting error:

[ERROR] You must set at least the property 'base.deploy.dir'!

This happens even when building only a single module.

--- a/README.md
+++ b/README.md
@@ -90,6 +90,17 @@ git clone https://github.com/eclipse-sw360/sw360.git
 cd sw360
 pip install pre-commit
 pre-commit install
+
+### Note on build requirements
+
+Please note that even partial or module-level Maven builds require deploy-related
+properties to be set due to enforced build rules.
+
+At a minimum, the `base.deploy.dir` property must be provided, otherwise the build
+will fail with a Maven Enforcer error.
+
+This applies even when building individual modules (for example, `libraries`).
+
 ```
 
 **Step 2**: Build the code

